### ISSUE TITLE
Is there packet data copy in f-stack if using recv/send ？

### ISSUE THREAD

**Issue by hubblezzhu** [F-Stack/f-stack#482]
Hi，since issue #467 says that using write/read will do memory copy. does using recv/send have the same problem？

any help will be approciated.

**Comment by jfb8856606** [comment]
Yes, `recv/send` same as `write/read`.

**Comment by jfb8856606** [comment]
Yes, `ff_recv/ff_send/ff_read/ff_write` all involve one memory copy from kernel mbuf to user buffer, same as the standard Linux socket API. This is by design.

For zero-copy operation, F-Stack provides `ff_zc_mbuf_get/ff_zc_mbuf_read/ff_zc_mbuf_write` APIs that allow direct access to mbuf data without copying. See `ff_api.h` for details.

Closing as resolved.
